src/main/java/com/arpnetworking/metrics/impl/ApacheHttpSink.java

```diff
@@ -123,10 +130,30 @@ private ApacheHttpSink(final Builder builder) {
     private static final Header CONTENT_TYPE_HEADER = new BasicHeader(HttpHeaders.CONTENT_TYPE, "application/octet-stream");
     private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(ApacheHttpSink.class);
 
+    /* package private */ static final class SingletonSupplier<T> implements Supplier<T> {
+
+        /* package private */ SingletonSupplier(final Supplier<T> supplier) {
+            _supplier = supplier;
+        }
+
+        @Override
+        public synchronized T get() {
```

Comment:
I don't particularly like the synchronized get here.  Seems like you could just gate the creation of the background HttpDispatch threads by the creation of the PoolingHttpClientConnectionManager.  Seems more straight forward.